opengrok-indexer/src/main/java/org/opengrok/indexer/history/Repository.java

```diff
@@ -381,39 +381,41 @@ final void createCache(HistoryCache cache, String sinceRevision)
         File directory = new File(getDirectoryName());
 
         History history;
-        try {
+        if (!(this instanceof RepositoryWithPerPartesHistory)) {
```

Comment:
I think this `instanceof` could be avoided with method overriding. The `RepositoryWithPerPartesHistory` could override a default implementation in `Repository`.

Reply:
good idea, that should work.